userspace/libscap/scap.h

```diff
@@ -324,9 +324,9 @@ typedef struct _scap_machine_info
 	uint64_t max_pid; ///< Highest PID number on this machine
 	char hostname[128]; ///< The machine hostname
 	uint64_t boot_ts_epoch; ///< Host boot ts in nanoseconds (epoch)
-	uint64_t reserved2; ///< reserved for future use
-	uint64_t reserved3; ///< reserved for future use
-	uint64_t reserved4; ///< reserved for future use
+	uint64_t self_pid_start_ts; ///< Agent start ts in nanoseconds (epoch)
```

Comment:
Sounds good, but I miss the context on why these were reserved. Maybe because of padding and backward compatibility with something?

My only concern, and broader question, is: will we be able to further grow this struct in the future, when we need more info?

Reply:
AFAIK it's for future extensibility (so that we don't need another hack while reading scap files). Note that this patch breaks compatibility already (the old/new sizes are very much different) so we need changes to savefile code in this PR.

Personally, I'd rather:
* add the string field _after_ the currently reserved fields (let's keep the reserved ones for future integer vars)
* use an `uint64_t flags` instead of a bool and `#define` a numeric constant for the one bit we need so far (BTW we can't use `bool` anyway since its size [isn't guaranteed to be the same](https://stackoverflow.com/a/4897859/922138) on all platforms and we save this struct directly to disk)

(deep in my patch queue there's another patch that adds stuff to machine_info; it needs just a few bits, like 3 or 4 I think, so maybe it could piggyback on the flags field without adding yet another one and using up a pretty scarce resource)